hmda-analytics/src/main/scala/hmda/analytics/HmdaAnalyticsApp.scala

```diff
@@ -66,118 +66,105 @@ object HmdaAnalyticsApp extends App with TransmittalSheetComponent with LarCompo
    * In beta namespace this environment variable has currentSchema=hmda_beta_user appended to it to change the schema
    * to BETA
    */
-  val tsTableName2018  = config.getString("hmda.analytics.2018.tsTableName")
-  val larTableName2018 = config.getString("hmda.analytics.2018.larTableName")
+
+  case class YearlyTransmittalSheetRepositoryWrapper(year: String){
+    val transmittalSheet = new TransmittalSheetRepository(dbConfig, getTableNameByYear(year))
+
+    def getTransmittalSheet = transmittalSheet
+
+    def getId(row: TransmittalSheetTable): Unit = transmittalSheet.getId(row)
+    def createSchema(): Unit = transmittalSheet.createSchema()
+    def dropSchema(): Unit = transmittalSheet.dropSchema()
+
+    def insert(ts: TransmittalSheetEntity): Future[Int] = transmittalSheet.insert(ts)
+
+    def findByLei(lei: String): Future[Seq[TransmittalSheetEntity]] = transmittalSheet.findByLei(lei)
+
+    def deleteByLei(lei: String): Future[Int] = transmittalSheet.deleteByLei(lei)
+
+    def deleteByLeiAndQuarter(lei: String): Future[Int] = transmittalSheet.deleteByLeiAndQuarter(lei)
+
+    def updateByLei(ts: TransmittalSheetEntity): Future[Int] = transmittalSheet.updateByLei(ts)
+
+    def count(): Future[Int] = transmittalSheet.count()
+
+    private def getTableNameByYear(year: String): String = {
+      val configString = s"hmda.analytics.$year.tsTableName"
+      config.getString(configString)
+    }
+
+  }
+
+  case class QuarterlyTransmittalSheetRepositoryWrapper(year: String, quarter: String){
+    val transmittalSheet = new TransmittalSheetRepository(dbConfig, getTableNameByYear(year))
+
+    def getTransmittalSheet = transmittalSheet
+
+
+    def getId(row: TransmittalSheetTable): Unit = transmittalSheet.getId(row)
+    def createSchema(): Unit = transmittalSheet.createSchema()
+    def dropSchema(): Unit = transmittalSheet.dropSchema()
+
+    def insert(ts: TransmittalSheetEntity): Future[Int] = transmittalSheet.insert(ts)
+
+    def findByLei(lei: String): Future[Seq[TransmittalSheetEntity]] = transmittalSheet.findByLei(lei)
+
+    def deleteByLei(lei: String): Future[Int] = transmittalSheet.deleteByLei(lei)
+
+    def deleteByLeiAndQuarter(lei: String): Future[Int] = transmittalSheet.deleteByLeiAndQuarter(lei)
+
+    def updateByLei(ts: TransmittalSheetEntity): Future[Int] = transmittalSheet.updateByLei(ts)
+
+    def count(): Future[Int] = transmittalSheet.count()
+
+    private def getTableNameByYear(year: String): String = {
```

Comment:
Looks like the getTableNameByYear functions in the quarterly wrappers take only the year as a function arg but get the quarter through the constructor, which is the class body. This is confusing. Recommend to standardize the getTableNameByYear functions or just remove them.

Reply:
I removed the paramters to just make them simple private helpers